stable_baselines3/common/vec_env/vec_normalize.py

```diff
@@ -159,7 +160,11 @@ def set_venv(self, venv: VecEnv) -> None:
         """
         if self.venv is not None:
             raise ValueError("Trying to set venv of already initialized VecNormalize wrapper.")
-        VecEnvWrapper.__init__(self, venv)
+        self.venv = venv
+        self.num_envs = venv.num_envs
+        venv.observation_space = self.observation_space
+        venv.action_space = self.action_space
+        self.class_attributes = dict(inspect.getmembers(self.__class__))
```

Comment:
why is that needed?

Reply:
```python
VecEnvWrapper.__init__(self, venv) 
```

is equivalent to

```python
self.venv = venv
self.num_envs = venv.num_envs
self.observation_space = venv.observation_space
self.action_space = self.action_space
self.class_attributes = dict(inspect.getmembers(self.__class__))
```

The only change in this PR is that we change 

```python
self.observation_space = venv.observation_space
self.action_space = self.action_space
```

to 

```python
venv.observation_space = self.observation_space
venv.action_space = self.action_space
```

Then:

```python
self.venv = venv  # is needed because the environment is not set
self.num_envs = venv.num_envs  # is needed because the loaded env, and the one to set don't necessarily share the number of env, see the unit test.
venv.observation_space = self.observation_space  # see other comment
venv.action_space = self.action_space  # not sure if it's needed, but it could lead to the same break for the same reason
self.class_attributes = dict(inspect.getmembers(self.__class__))  # not sure if its need, but it keeps consistency with the old implementation
```

Reply:
https://github.com/DLR-RM/stable-baselines3/blob/489b1fdaf2958ba5d63ed5a90f20540cd4107b13/stable_baselines3/common/vec_env/base_vec_env.py#L249-L262

And then `VecEnv.__init__`:

https://github.com/DLR-RM/stable-baselines3/blob/489b1fdaf2958ba5d63ed5a90f20540cd4107b13/stable_baselines3/common/vec_env/base_vec_env.py#L59-L62

Reply:
oh, I forgot about `self.class_attributes = dict(inspect.getmembers(self.__class__))`, yes we need that when doing recursion.